### ISSUE TITLE
brew cleanup should be folded in

### ISSUE THREAD

**Issue by alper** [Homebrew/brew#2916]
I don't understand the use case for `brew cleanup` and why homebrew leaves behind old packages if I never use them anyway.

Lots of people are surprised when they discover the cleanup command and then see that it returns them multiple gigabytes of space. Shouldn't the default action be for homebrew to cleanup after itself?

**Comment by MikeMcQuaid** [comment]
> Shouldn't the default action be for homebrew to cleanup after itself?

Yes, however, it's too slow to do that automatically or even check if it's needed because we don't have a metadata store. It's a desired feature if you or anyone else would want to work on it, though.

**Comment by chdiza** [comment]
> I never use them anyway.

But others do.

I only run `cleanup` periodically, because there are good reasons to keep older kegs installed.  E.g., if I just installed `foo` version 1.9, but also I still have the 1.8 keg around, I might very well *want* the 1.8 keg around in case something is wrong with 1.9.  Then I can `brew switch` and get a working `foo` back.  I've needed to do this many, many times.

In my experience running `cleanup` after every install is just asking for trouble, and should definitely not be made into any kind of default.

**Comment by alper** [comment]
OK. Maybe it shouldn't be folded in, but the core issue is that some users are unpleasantly surprised when they first find out how much space they could have saved. 

Would it be feasible to calculate the savings and print those out after update/upgrade regularly if this is over a certain threshold (1G)? If so, I'd close this and open that as another issue.

**Comment by MikeMcQuaid** [comment]
We would need something like https://github.com/Homebrew/brew/issues/568 to be implemented first before either folding in or printing a warning are possible without it being very slow.

**Comment by alper** [comment]
OK. That is a bit beyond my Ruby skill level, but just in general: since something like `update` is network bound anyway, could it work to run local statistics in parallel and display the result if it has returned when `update` is done?
